REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/security/CustomPreAuthorize.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.esm.security;
+
+import com.epam.esm.security.exception.SecurityAccessDeniedException;
+import com.epam.esm.user.entity.User;
+import com.epam.esm.user.entity.roles.Role;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service("securityService")
+@Slf4j
+public class CustomPreAuthorize {
+    public boolean customFilter(String id) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
Naming of this class and method is confusing to me. Rename please to more suitable names to its purpose

Reply:
Renamed as 'SecurityAccessService' class and 'hasAccess' method